cmds/core/kill/kill.go

```diff
@@ -22,27 +22,27 @@ package main
 
 import (
 	"fmt"
+	"io"
+	"log"
 	"os"
 )
 
 const eUsage = "Usage: kill -l | kill [<-s | --signal | -> <signame|signum>] pid [pid...]"
 
-func usage() {
-	die(eUsage)
+func usage(w io.Writer) {
+	fmt.Fprintf(w, "%s\n", eUsage)
 }
 
-func die(msg string, args ...interface{}) {
-	fmt.Fprintf(os.Stderr, msg, args...)
-	fmt.Fprintf(os.Stderr, "\n")
-	os.Exit(1)
-}
-
-func main() {
-	op := os.Args[1]
-	pids := os.Args[2:]
+func killProcess(args []string, w io.Writer) error {
```

Comment:
the variadic would still work
killProcess(w io.Writer, args ...string)
len(args) test would still work.

Your call on this one, but I'd still pefer the io.Writer as the first arg, and variadic form.